src/ios/BranchSDK.m

```diff
@@ -84,31 +84,30 @@ - (void)initSession:(CDVInvokedUrlCommand*)command
         BOOL isFromUniversalLink = NO;
         NSString *resultString = nil;
         CDVPluginResult *pluginResult = nil;
-
+        
         // NOTE: For Universal Links. Using clicked_branch_link key as condition at the moment to identify if block is run due to Universal Links.
         isFromUniversalLink = [[params objectForKey:@"+clicked_branch_link"] boolValue];
-
+        
         if (!error) {
-            if (params != nil && [params count] > 0) {
+            if (params != nil && [params count] > 0 && isFromUniversalLink) {
```

Comment:
if isFromUniversalLink is within the if statement (instead of being a part of this one), we'll run into the case where params contains +clicked_branch_link, so the first two parts of this if are true, but then isFromUniversalLink is false, and then we'd never resolve the promise, so I moved the isFromUniversalLink check up here.